token/program-2022/src/state.rs

```diff
@@ -4,12 +4,50 @@ use crate::instruction::MAX_SIGNERS;
 use arrayref::{array_mut_ref, array_ref, array_refs, mut_array_refs};
 use num_enum::TryFromPrimitive;
 use solana_program::{
+    clock::Epoch,
     program_error::ProgramError,
     program_option::COption,
     program_pack::{IsInitialized, Pack, Sealed},
     pubkey::Pubkey,
 };
 
+/// Different kinds of accounts. Note that `Mint`, `Account`, and `Multisig` types
+/// are determined exclusively by the size of the account, and are not included in
+/// the account data. These `AccountType`s are only included if mixins have been
+/// initialized.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AccountType {
+    /// Marker for 0 data
+    Uninitialized,
+    /// Mint account with additional mixins
+    Mint,
+    /// Token holding account with additional mixins
+    Account,
+}
+
+impl Default for AccountType {
+    fn default() -> Self {
+        Self::Uninitialized
+    }
+}
+
+/// Mixins that can be applied to mints or accounts.  Mint mixins must only be
```

Comment:
Ok!  Time to open this can of worms since we're pretty much all done with the important stuff.  Do we want to keep the term "Mixin" or use something more watered down like "Attribute", "Trait", or "Extension"?

Reply:
I've been liking `Attribute` most, and `Extension` is good too.  `Trait` is a loaded word that we may want to avoid.  `Feature` is the only word I can come up with, but that's also loaded.

People seem to have a bizarre relationship with the word mixin. Ex-Django devs?

Reply:
I've only been following this PR lightly, but fwiw I like Trait or Extension for this.

Reply:
Welp that leaves only Extension in the venn diagram

Reply:
Works for me!